contracts/integrations/test/simulation/stake_management_test.ts

```diff
@@ -0,0 +1,53 @@
+import { BlockchainBalanceStore } from '@0x/contracts-exchange';
+import { blockchainTests } from '@0x/contracts-test-utils';
+import * as _ from 'lodash';
+
+import { Staker } from '../actors';
+import { DeploymentManager } from '../utils/deployment_manager';
+import { AssertionResult } from '../utils/function_assertions';
+
+import { PoolManagementSimulation } from './pool_management_test';
+import { Simulation, SimulationEnvironment } from './simulation';
+
+export class StakeManagementSimulation extends Simulation {
+    protected async *_assertionGenerator(): AsyncIterableIterator<AssertionResult | void> {
+        const { deployment, balanceStore } = this.environment;
+        const poolManagement = new PoolManagementSimulation(this.environment);
+
+        const staker = new Staker({ name: 'Staker', deployment, simulationEnvironment: this.environment });
+        await staker.configureERC20TokenAsync(deployment.tokens.zrx);
+        balanceStore.registerTokenOwner(staker.address, staker.name);
+
+        const actions = [
+            staker.simulationActions.validStake,
+            staker.simulationActions.validUnstake,
+            staker.simulationActions.validMoveStake,
+            poolManagement.generator,
+        ];
```

Comment:
I'm a big fan of this yield pattern. It's a lot like using closures for setup, but it seems like it will not bloat the heap as much.